src/hugr/rewrite/simple_replace.rs

```diff
@@ -505,15 +496,27 @@ pub(in crate::hugr::rewrite) mod test {
             .collect_vec();
         let inputs = h
             .node_outputs(input)
-            .filter(|&p| h.get_optype(input).signature().get(p).is_some())
+            .filter(|&p| {
+                h.get_optype(input)
+                    .signature()
+                    .unwrap()
+                    .port_type(p)
+                    .is_some()
```

Comment:
Perhaps an `OpType::df_port_type` helper would be useful?

Alternatively,
```suggestion
                h.signature(input).is_some_and(|s| s.port_type(p).is_some())
```
(there's many other cases of `hugr.get_optype(n).signature()`).

Reply:
I've taken a different approach - casting to known types and using
`dataflow_signature` rather than continuing to wrap `FunctionType` methods on to `OpType`